tests/ui/test_ui_tools.py

```diff
@@ -1217,6 +1218,105 @@ def test_main_view_generates_EDITED_label(self, mocker,
+            'type': 'stream',
+            'is_me_message': False,
+            'sender_short_name': 'aaron',
+            'flags': ['read'],
+            'sender_email': 'AARON@zulip.com'
+        }),
+        ({
+            'sender_id': 1,
+            'display_recipient': [{
+                'is_mirror_dummy': False,
+                'short_name': 'aaron',
+                'email': 'AARON@zulip.com',
+                'id': 1,
+                'full_name': 'aaron'
+            }, {
+                'is_mirror_dummy': False,
+                'short_name': 'iago',
+                'email': 'iago@zulip.com',
+                'id': 5,
+                'full_name': 'Iago'
+            }],
+            'sender_full_name': 'Iago',
+            'submessages': [],
+            'subject': '',
+            'id': 107,
+            'subject_links': [],
+            'recipient_id': 1,
+            'content': '<p>what are you planning to do this week</p>',
+            'timestamp': 1532103879,
+            'sender_realm_str': 'zulip',
+            'client': 'ZulipTerminal',
+            'content_type': 'text/html',
+            'reactions': [],
+            'type': 'private',
+            'is_me_message': False,
+            'sender_short_name': 'iago',
+            'flags': ['read'],
+            'sender_email': 'iago@zulip.com'
+        })])
+    def test_keypress_edit_message(self, mocker, message):
+        key = 'e'
+        size = (20,)
+        msg_box = MessageBox(message, self.model, message)
+        msg_box.model.user_id = message['sender_id']
+        msg_box.model.initial_data = dict()
+        msg_box.model.initial_data['realm_allow_message_editing'] = False
+        msg_box.model.initial_data[
+            'realm_message_content_edit_limit_seconds'] = 60
+        msg_box.model.client.get_raw_message.return_value = {
+            'raw_content': "MSG"
+        }
+        write_box = msg_box.model.controller.view.write_box
+        write_box.is_edit_mode = False
+        # Editing not allowed
+        msg_box.keypress(size, key)
+
+        write_box.msg_write_box.set_edit_text.assert_not_called()
+        assert write_box.is_edit_mode is False
+
+        # Time limit exceeded
+        msg_box.model.initial_data['realm_allow_message_editing'] = True
+        msg_box.keypress(size, key)
+
+        write_box.msg_write_box.set_edit_text.assert_not_called()
+        assert write_box.is_edit_mode is False
+
+        # MSG SENT BY OTHER USER
+        msg_box.model.user_id = 2  # != sender_id
+        msg_box.keypress(size, key)
+
+        write_box.msg_write_box.set_edit_text.assert_not_called()
+        assert write_box.is_edit_mode is False
+
+        # MEET ALL THE CONDITIONS
+        msg_box.message['timestamp'] = time.time()
+        msg_box.model.user_id = message['sender_id']
+        msg_box.keypress(size, key)
+
+        write_box.msg_write_box.set_edit_text.assert_called_once_with("MSG")
+        assert write_box.is_edit_mode is True
+
```

Comment:
Whenever I see these sections within one test, I wonder whether it could be 'parametrize'd. I think this would make it clearer. With `id`s, it could be even clearer, particularly if a test subsequently fails.